esp-hal/src/rtc_cntl/mod.rs

```diff
@@ -1278,7 +1278,7 @@ pub fn wakeup_cause() -> SleepSource {
 
 // libphy.a can pull this in on some chips, we provide it here in the hal
 // so that either ieee or esp-wifi gets it for free without duplicating in both
-#[unsafe(no_mangle)]
+#[cfg_attr(feature = "rt", unsafe(no_mangle))]
```

Comment:
I think we should remove this, and put it back in esp-wifi, I don't like the idea of exporting rando symbols from the hal

Reply:
we can make sure we do what esp-idf does: storing the xtal freq in one of the RTC retention registers (RTC_CNTL_RTC_STOREx_REG - don't remember which one) - then we can put this function into esp-rom-sys 🤔

Reply:
Okay, let's do that. I'll file an issue.